src/interfaces.ts

```diff
@@ -411,6 +411,22 @@ export interface FileLink<T> {
 
 export interface GetTypeOptions {
     flags: SymbolTypeFlags;
-    typeChain?: TypeResolution[];
+    typeChain?: TypeChainEntry[];
     //TODO: Add a TypeCache that can be used to look up and store types to short circuit reference type look ups
 }
+
+export class TypeChainEntry {
+    constructor(public name: string, public type: BscType, public range: Range) {
+    }
+    get isResolved() {
+        return this.type?.isResolvable();
+    }
+}
+
+export interface TypeChainProcessResults {
```

Comment:
```suggestion
export interface TypeChainProcessResult {
```
Results makes me think there are multiple "results", when really this represents a single "result" just with multiple properties.